ibmdbpy/base.py

```diff
@@ -565,22 +565,29 @@ def exists_model(self, modelname):
         """
         modelname = ibmdbpy.utils.check_tablename(modelname)
         if '.' in modelname:
+            modelschema = modelname.split('.')[0]
             modelname = modelname.split('.')[-1]
+        else:
+            modelschema = self.current_schema
 
-        data = self.show_models()
+        # check if schema exists to avoid exception thrown by idax.list_models
+        schemaquery = "select count(*) from syscat.schemata where schemaname = '" + modelschema + "'"
+        schemaexists = self.ida_scalar_query(schemaquery) >= 1
+        if not schemaexists:
+          return False
+
+        data = self.ida_query("call idax.list_models('schema=%s, where=MODELNAME=''%s''')" % (modelschema, modelname))
 
         if not data.empty:
-            if modelname in data['MODELNAME'].values:
-                return True
+            return True
 
         tablelist = self.show_tables(show_all=True)
+        tablelist= tablelist[(tablelist['TABSCHEMA']==modelschema) & (tablelist['TABNAME']==modelname)]
         if len(tablelist):
-            if modelname in tablelist['TABNAME'].values:
-                tabletype = tablelist[tablelist['TABNAME'] == modelname]['TYPE'].values[0]
-                raise TypeError("%s exists but is not a model (of type '%s')"
-                                %(modelname, tabletype))
-            else:
-                return False
+             tabletype = tablelist['TYPE'].values[0]
+             raise TypeError("%s.%s exists, but is not a model (of type '%s')"
+                  %(modelschema, modelname, tabletype))
```

Comment:
Please keep the alignment for readability:

```suggestion
             raise TypeError("%s.%s exists, but is not a model (of type '%s')"
                             % (modelschema, modelname, tabletype))
```